archivist/sboms.py

```diff
@@ -0,0 +1,164 @@
+"""SBOMS interface
+
+   Direct access to the sboms endpoint
+
+   The user is not expected to use this class directly. It is an attribute of the
+   :class:`Archivist` class.
+
+   For example instantiate an Archivist instance and execute the methods of the class:
+
+   .. code-block:: python
+
+      with open(".auth_token", mode="r", encoding="utf-8") as tokenfile:
+          authtoken = tokenfile.read().strip()
+
+      # Initialize connection to Archivist
+      arch = Archivist(
+          "https://app.rkvst.io",
+          auth=authtoken,
+      )
+      with open("bom.xml") as fd:
+          sbom = arch.sboms.upload(fd)
+
+"""
+
+# pylint:disable=too-few-public-methods
+
+from typing import BinaryIO, Dict, Optional
+from copy import deepcopy
+
+import logging
+
+from requests.models import Response
+
+# pylint:disable=unused-import      # To prevent cyclical import errors forward referencing is used
+# pylint:disable=cyclic-import      # but pylint doesn't understand this feature
+from . import archivist as type_helper
+
+from .constants import SBOMS_SUBPATH, SBOMS_LABEL, SBOMS_METADATA, SBOMS_WILDCARD
+from .dictmerge import _deepmerge
+from .sbommetadata import SBOM
+
+#: Default page size - number of entities fetched in one REST GET in the
+#: :func:`~_AssetsClient.list` method. This can be overridden but should rarely
+#: be changed.
+DEFAULT_PAGE_SIZE = 500
```

Comment:
Minor: this needs to be synced up with the max page size which is tbc within the api, so we best keep an eye on both!

Reply:
I am going to leave this for now as it applies to other endpoints as well..
Correct solution is have a default of None

Reply:
sounds good!

Reply:
Raised issue here https://github.com/jitsuin-inc/archivist-python/issues/72